### PR TITLE
docs: update sysmon to oshisysmon in reference docs

### PR DESCRIPTION
**Pull request by 317brian**
https://github.com/apache/druid/pull/14359 in 27.0.0 added OshiSysMonitor support to Druid, and Druid deprecated support for SysMonitor. 

The reference docs still refer to SysMonitor. Since we didn't provide any additional migration steps beyond saying to include the correct module, I think we only need to update the config names in the reference page.

If this change isn't needed and the config name is staying the same, we should probably note that instead so people don't get confused.

### CODE REVIEW

**Comment by 317brian** [comment]
@hardikbajaj would you mind taking a look and confirming that this change is needed since you opened the PR that added Oshi.

**Comment by hardikbajaj** [comment]
Hey @317brian , the updated Documentation for OshiSysMonitor is merged and added in 28.0 milestone https://github.com/apache/druid/pull/14912. I think i missed to add these changes in that PR, it would be needed, but can we keep config for both SysMonitor and OshiSysMonitor, and mark SysMonitor as deprecated. We're doing this everywhere.

**Comment by 317brian** [comment]
Fixed. PTAL @hardikbajaj

File: docs/configuration/index.md [hunk 1]
```diff
@@ -383,7 +383,8 @@ Metric monitoring is an essential part of Druid operations.  The following monit
 |Name|Description|
 |----|-----------|
 |`org.apache.druid.client.cache.CacheMonitor`|Emits metrics (to logs) about the segment results cache for Historical and Broker processes. Reports typical cache statistics include hits, misses, rates, and size (bytes and number of entries), as well as timeouts and and errors.|
-|`org.apache.druid.java.util.metrics.SysMonitor`|Reports on various system activities and statuses using the [SIGAR library](https://github.com/hyperic/sigar). Requires execute privileges on files in `java.io.tmpdir`. Do not set `java.io.tmpdir` to `noexec` when using `SysMonitor`.|
+|`org.apache.druid.java.util.metrics.SysMonitor`| Deprecated. Use OshiSysMonitor instead. Reports on various system activities and statuses using the [SIGAR library](https://github.com/hyperic/sigar). Requires execute privileges on files in `java.io.tmpdir`. Do not set `java.io.tmpdir` to `noexec` when using `SysMonitor`.|
```
Comment (ektravel) [inline docs/configuration/index.md:386]:
```suggestion
|`org.apache.druid.java.util.metrics.SysMonitor`| Deprecated. Use OshiSysMonitor instead. Reports on various system activities and statuses using the [SIGAR library](https://github.com/hyperic/sigar). Requires execute privileges on files in `java.io.tmpdir`. Do not set `java.io.tmpdir` to `noexec` when using `SysMonitor`.|
```
Why is SysMonitor in code font and OshiSysMonitor is not?

**Review by ektravel — APPROVED** [review]
One question regarding code font. Otherwise, looks good.

File: docs/configuration/index.md (hunk 1 above)
Comment (317brian) [inline docs/configuration/index.md:386]:
```suggestion
|`org.apache.druid.java.util.metrics.SysMonitor`| Deprecated. Use OshiSysMonitor instead. Reports on various system activities and statuses using the [SIGAR library](https://github.com/hyperic/sigar). Requires execute privileges on files in `java.io.tmpdir`. Do not set `java.io.tmpdir` to `noexec` when using SysMonitor.|
```

**Comment by 317brian** [comment]
@@hardikbajaj PTAL

**Review by hardikbajaj — COMMENTED** [review]
@317brian One miner nit, rest LGTM. @abhishekagarwal87 can you also have a look here if this looks good to you

**Comment by 317brian** [comment]
> @317brian One miner nit, rest LGTM. @abhishekagarwal87 can you also have a look here if this looks good to you

Cool, thanks for taking a look! I think GitHub ate your feedback for the nit though. I don't see it 😬

**Comment by 317brian** [comment]
Ping @abhishekagarwal87

File: docs/configuration/index.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ There are four JVM parameters that we set on all of our services:
      - The temp directory should not be volatile tmpfs.
      - This directory should also have good read and write speed.
      - Avoid NFS mount.
-     - The `org.apache.druid.java.util.metrics.SysMonitor` requires execute privileges on files in `java.io.tmpdir`. If you are using the system monitor, do not set `java.io.tmpdir` to `noexec`.
+     - The `org.apache.druid.java.util.metrics.OshiSysMonitor` and `org.apache.druid.java.util.metrics.SysMonitor` both require execute privileges on files in `java.io.tmpdir`. If you are using the system monitor, do not set `java.io.tmpdir` to `noexec`. Note that SysMonitor is deprecated. We recommend using OshiSysMonitor.
```
Comment (abhishekagarwal87) [inline docs/configuration/index.md:109]: @hardikbajaj - do you know if the OshiSysMonitor needs this privilege?

Comment (317brian) [reply docs/configuration/index.md:109]: @hardikbajaj ping

**Pull request closed without merging**